### PR TITLE
[Bug fix] Qwen3-VL: slice page table before multimodal prefill

### PR DESCRIPTION
**Pull request by viktorpusTT**
### PR Category
Bug fix

### Summary
Qwen3-VL's multimodal prefill processes one user at a time, but it passed vLLM's full persistent page table into the non-chunked single-user prefill path. That table is sized for the request's maximum block capacity, so columns after the allocated prompt blocks are zero-filled padding.

The inherited `prefill_forward_single_user_text` path does not slice the table during normal, non-chunked prefill. It forwards the full table into `prepare_inputs_prefill` and `ttnn_prefill_forward`. The prefill attention path uses the table width to determine how much K/V state to write, so the padded zero entries direct those extra writes to physical KV block 0. This can overwrite valid prompt K/V in block 0 before decode starts.

The old Qwen3-VL comment said the parent method handled page-table slicing, but that is only true for the chunked or prefix-cached branch. The normal prefill branch [passes the supplied table through unchanged](https://github.com/tenstorrent/tt-metal/blob/87a43f86f1620e5f5ec967ae5febba7987f24345/models/tt_transformers/tt/generator.py#L1235-L1251).

This change slices the page table to the blocks required by `decoding_pos` before calling the single-user prefill method. That matches the established pattern in:

- [Qwen2.5-VL](https://github.com/tenstorrent/tt-metal/blob/87a43f86f1620e5f5ec967ae5febba7987f24345/models/demos/qwen25_vl/tt/generator.py#L58-L69)
- [the common model executor](https://github.com/tenstorrent/tt-metal/blob/87a43f86f1620e5f5ec967ae5febba7987f24345/models/common/models/executor.py#L891-L900)
- [the shared TT transformer generator](https://github.com/tenstorrent/tt-metal/blob/87a43f86f1620e5f5ec967ae5febba7987f24345/models/tt_transformers/tt/generator.py#L766-L779)

A focused unit test verifies that Qwen3-VL calls `_get_prefill_user_page_table` with the actual multimodal decoding position and forwards only the returned slice.

### Notes for reviewers
The behavioral change is limited to Qwen3-VL multimodal prefill. Decode page-table handling is unchanged.

Validation:
- `python3 -m py_compile models/demos/qwen3_vl/tt/generator_vllm.py models/demos/qwen3_vl/tests/test_generator_vllm.py`
- `git diff --check origin/main...HEAD`
- Focused pytest was not run locally because the available system Python does not have pytest installed (`No module named pytest`).
- CI: https://github.com/tenstorrent/tt-metal/actions/runs/30254783587

Made with [Cursor](https://cursor.com)

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:viktorpus/fix-qwen3-vl-prefill-page-table-main)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:viktorpus/fix-qwen3-vl-prefill-page-table-main)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-sanity-tests.yaml?query=branch:viktorpus/fix-qwen3-vl-prefill-page-table-main)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:viktorpus/fix-qwen3-vl-prefill-page-table-main)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:viktorpus/fix-qwen3-vl-prefill-page-table-main)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:viktorpus/fix-qwen3-vl-prefill-page-table-main)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:viktorpus/fix-qwen3-vl-prefill-page-table-main)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:viktorpus/fix-qwen3-vl-prefill-page-table-main)

### CODE REVIEW

File: models/demos/qwen3_vl/tests/test_generator_vllm.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# SPDX-FileCopyrightText: © 2026 Tenstorrent USA, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+
+from models.demos.qwen3_vl.tt import generator_vllm
```
[the remark this answers is not in the document]
Comment (viktorpusTT) [reply models/demos/qwen3_vl/tests/test_generator_vllm.py:9]: @copilot `generator_vllm.py` is part of this repo and this very PR. Re-check your finding.

**Review by gwangTT — APPROVED** [review]
